From 74eb2c42cabb91d1fdc5fd30a9ac9596cad1f160 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 9 Oct 2014 21:00:26 -0400
Subject: [fs] xfs: fix uflags detection at xfs_fs_rm_xquota

Message-id: <1412888441-4181-33-git-send-email-sandeen@redhat.com>
Patchwork-id: 97340
O-Subject: [RHEL7.1 PATCH 32/47] [fs] xfs: fix uflags detection at xfs_fs_rm_xquota
Bugzilla: 1145837
RH-Acked-by: Brian Foster <bfoster@redhat.com>
RH-Acked-by: Dave Chinner <dchinner@redhat.com>

Bugzilla: 1145837
Upstream Status: Committed upstream
Build Info: https://brewweb.devel.redhat.com/taskinfo?taskID=8082426
Tested: locally using xfstests

Thanks,
-Eric

    commit 74dc93a9087fc71240486d914b4a95f8e510e0e4
    Author: Jie Liu <jeff.liu@oracle.com>
    Date:   Thu Jul 24 21:27:17 2014 +1000

    xfs: fix uflags detection at xfs_fs_rm_xquota

    We are intended to check up uflags against FS_PROJ_QUOTA rather than
    FS_USER_UQUOTA once more, it looks to me like a typo, but might cause
    the project quota metadata space can not be removed.

    Signed-off-by: Jie Liu <jeff.liu@oracle.com>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Dave Chinner <david@fromorbit.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/xfs/xfs_quotaops.c b/fs/xfs/xfs_quotaops.c
index 2ad1b98..a3d3058 100644
--- a/fs/xfs/xfs_quotaops.c
+++ b/fs/xfs/xfs_quotaops.c
@@ -123,7 +123,7 @@ xfs_fs_rm_xquota(
   flags |= XFS_DQ_USER;
  if (uflags & FS_GROUP_QUOTA)
   flags |= XFS_DQ_GROUP;
- if (uflags & FS_USER_QUOTA)
+ if (uflags & FS_PROJ_QUOTA)
   flags |= XFS_DQ_PROJ;
 
  return -xfs_qm_scall_trunc_qfiles(mp, flags);
-- 
1.7.1